addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightCommunicationV3.java

```diff
@@ -0,0 +1,645 @@
+
+        if (bulb != 5) {
+            state.ledMode = 0;
+        }
+
+        if (mode > state.ledMode) {
+            int repeatCount = (mode - state.ledMode) / 10;
+            for (int i = 0; i < repeatCount; i++) {
+
+                nextAnimationMode(bulb, state);
+            }
+        } else if (mode < state.ledMode) {
+            int repeatCount = (state.ledMode - mode) / 10;
+            for (int i = 0; i < repeatCount; i++) {
+
+                previousAnimationMode(bulb, state);
+            }
+        }
+
+        state.ledMode = mode;
+    }
+
+    @Override
+    public void previousAnimationMode(int bulb, MilightThingState state) {
+        logger.debug("milight: sendDiscoModeDown");
+        byte messageBytes[] = { 0x28, 0x00, 0x55 };
+        setPower(bulb, true, state);
+
+        sendQueue.add(messageBytes, bulb, QueuedCategorizedSend.NO_CATEGORY, false);
+        state.ledMode = Math.min(state.ledMode - 10, 0);
+    }
+
+    @Override
+    public void nextAnimationMode(int bulb, MilightThingState state) {
+        logger.debug("milight: sendDiscoModeUp");
+        byte messageBytes[] = null;
+        if (bulb < 6) {
+            messageBytes = new byte[] { 0x27, 0x00, 0x55 };
+        }
+        if (bulb >= 6) {
+            messageBytes = new byte[] { 0x4D, 0x00, 0x55 };
+        }
+        setPower(bulb, true, state);
+
+        sendQueue.add(messageBytes, bulb, QueuedCategorizedSend.NO_CATEGORY, false);
+        state.ledMode = Math.max(state.ledMode + 10, 100);
+    }
+
+    @Override
+    public void setAnimationSpeed(int bulb, int speed, MilightThingState state) {
+        if (speed > state.animationSpeed) {
+            int repeatCount = (speed - state.animationSpeed) / 10;
+            for (int i = 0; i < repeatCount; i++) {
+
+                changeSpeed(bulb, 1, state);
+            }
+        } else if (speed < state.animationSpeed) {
+            int repeatCount = (state.animationSpeed - speed) / 10;
+            for (int i = 0; i < repeatCount; i++) {
+
+                changeSpeed(bulb, -1, state);
+            }
+        }
+
+        state.animationSpeed = speed;
+    }
+
+    @Override
+    public void changeSpeed(int bulb, int relative_speed, MilightThingState state) {
+        if (relative_speed > 0) {
+            logger.debug("milight: sendIncreaseSpeed");
+            byte messageBytes[] = null;
+            switch (bulb) {
+                case 5:
+                    // message increaseSpeed rgb bulbs
+                    messageBytes = new byte[] { 0x25, 0x00, 0x55 };
+                    break;
+                case 6:
+                case 7:
+                case 8:
+                case 9:
+                case 10:
+                    // message increaseSpeed rgb-w bulbs
+                    messageBytes = new byte[] { 0x44, 0x00, 0x55 };
+                    break;
+            }
+            setPower(bulb, true, state);
+
+            sendQueue.add(messageBytes, bulb, (byte) 0, true);
```

Comment:
For older bridges you were required to send the power on and then the actual command. I thought it wouldn't hurt if I'm doing it here as well, although the documentation doesn't require this.